### ISSUE TITLE
Using Python images improvement

### ISSUE THREAD

**Issue by rhcarvalho** [openshift/openshift-docs#1283]
tl;dr: Document what the assemble script of our S2I builder images (Python, Ruby, ...) are able to do.

Apparently nowhere in the docs ([main](https://docs.openshift.org/latest/using_images/s2i_images/python.html), [builder image readme](https://github.com/openshift/sti-python/blob/master/README.md)) we document that the default assemble scripts look for project dependencies in a file named `requirements.txt` or `setup.py`.

We also may omit other interesting bits of the default capabilities of our scripts, and that leave users wondering e.g. how do they install their project dependencies when using Source-To-Image builds.

Our S2I Python images automatically install dependencies on build, either by `pip install requirements.txt` or `python setup.py`.

Other builder images, like Ruby, Node.js, etc, have similar mechanisms to install user dependencies and we should document those.

**Comment by tnguyen-rh** [comment]
[BZ created](https://bugzilla.redhat.com/show_bug.cgi?id=1288488); closing.
